cmake/SetupDoxygen.cmake

```diff
@@ -14,25 +14,64 @@ if (DOXYGEN_FOUND)
   # Configure file that contains doxygen groups
   configure_file(docs/GroupDefs.hpp ${SPECTRE_DOXYGEN_GROUPS})
 
-  add_custom_target(
-      doc
-      COMMAND ${DOXYGEN_EXECUTABLE} ${PROJECT_BINARY_DIR}/docs/DoxyfileHtml 2>&1
-      DEPENDS
-      ${PROJECT_BINARY_DIR}/docs/DoxyfileHtml
-      ${SPECTRE_DOXYGEN_GROUPS}
-      )
-
-  add_custom_target(
-    doc-check
-    # search for warnings and print out 6 lines after to make sure
-    # majority of warning is printed so the results are actually
-    # useful.
-    COMMAND ! ${DOXYGEN_EXECUTABLE} ${PROJECT_BINARY_DIR}/docs/DoxyfileHtml 2>&1
-    | grep -A 6 'warning'
-    DEPENDS
-    ${PROJECT_BINARY_DIR}/docs/DoxyfileHtml
-    ${SPECTRE_DOXYGEN_GROUPS}
-    )
+  # We need Python for postprocessing the documentation
+  find_package(PythonInterp)
+  include(FindPythonModule)
+  find_python_module(bs4 FALSE)
+  find_python_module(pybtex FALSE)
+
+  if (PYTHONINTERP_FOUND AND PY_BS4 AND PY_PYBTEX)
+    add_custom_target(
+        doc
+        COMMAND
+        ${DOXYGEN_EXECUTABLE} ${PROJECT_BINARY_DIR}/docs/DoxyfileHtml 2>&1
+        && ${PYTHON_EXECUTABLE}
```

Comment:
You might be able to combine the majority of the code in the two branches by setting a variable like `set(DOXYGEN_POST_PROCESS_COMMAND "...")` and inserting that. I'm not certain, but I think it's worth a shot since it'll reduce the code almost in half :)